sdk/storage/azure-storage-blob/azure/storage/blob/aio/_download_async.py

```diff
@@ -46,8 +46,14 @@
 async def process_content(data: Any, start_offset: int, end_offset: int, encryption: Dict[str, Any]) -> bytes:
     if data is None:
         raise ValueError("Response cannot be None.")
-    await data.response.read()
-    content = cast(bytes, data.response.content)
+
+    if not data.response.is_stream_consumed:
+        content = b"".join([d async for d in data])
+    else:
+        # This happens when validate_content=True, the body is already read and the stream is closed.
+        await data.response.read()
+        content = cast(bytes, data.response.content)
```

Comment:
Refactored to `if-else`
As it turns out, this response object's `read()` method is smart enough to read from memory if it already exists. So calling `read()` is always safe, whether it is in memory or not.

So this code should be sufficient, the first code-path is so we can get around the auto-decompress setting `_decompress=True` on the response object, but using the iterator will use the generator's `decompress` kwarg (which we did set)

Reply:
The double load issue was from calling `.body()` (what this code did previously), so I guess that one is not smart enough